redis/client.py

```diff
@@ -232,6 +232,76 @@ def int_or_none(response):
     return int(response)
 
 
+def stream_list(response):
+    if response is None:
+        return None
+    return [(r[0], pairs_to_dict(r[1])) for r in response]
+
+
+def parse_recursive_dict(response):
+    if response is None:
+        return None
+    result = {}
+    while response:
+        k = response.pop(0)
+        v = response.pop(0)
+        if isinstance(v, list):
+            v = parse_recursive_dict(v)
+        result[k] = v
+    return result
+
+
+def parse_list_of_recursive_dicts(response):
+    if response is None:
+        return None
+    result = []
+    for group in response:
+        result.append(parse_recursive_dict(group))
+    return result
+
+
+def parse_xclaim(response):
+    if all(isinstance(r, (basestring, bytes)) for r in response):
+        return response
+    return stream_list(response)
+
+
+def parse_xreadgroup(response):
+    if response is None:
+        return None
+    return [[nativestr(r[0]), stream_list(r[1])] for r in response]
+
+
+def parse_xpending(response):
```

Comment:
Response callbacks can accept **options. These come from passing **kwargs to `execute_command`. In the `xtrim_range` function, you could pass a `parse_detail=True` to `execute_command`, then here say something like:

```python
if options.get('parse_detail', False):
    return parse_range_xpending(response)
```